packages/augur-sdk/src/state/db/AbstractDB.ts

```diff
@@ -43,15 +46,17 @@ export abstract class AbstractDB {
     }
   }
 
-  protected async upsertDocument(id: string, document: object): Promise<PouchDB.Core.Response> {
+  protected async upsertDocument(id: string, document: object): Promise<PouchDB.UpsertResponse> {
     const previousBlockRev = await this.getPouchRevFromId(id);
     const obj = Object.assign(
       previousBlockRev ? { _rev: previousBlockRev } : {},
       { _id: id },
       document
     );
 
-    return this.db.put(obj);
+    return this.db.upsert(id, () => {
```

Comment:
Do we still need all of the above logic since we're letting a plugin handle the revision / update machinery?

Reply:
@nuevoalex Fixing. Good catch. Thanks!